File: 프로그래머스/lv3/43238. 입국심사/입국심사.py
def solution(n, times):
    answer = int(1e18)
    times.sort()
    start = 1
    end = n* times[-1]

    while start <= end:
        mid = (start + end) // 2
        cnt = 0
        #print('mid',mid)

        for t in times:
            cnt += mid // t

        #print(cnt)

        if cnt >= n:
            end = mid - 1
            answer = min(answer, mid)
        elif cnt < n:
            start = mid + 1

    return answer

File: 프로그래머스/3/43238. 입국심사/입국심사.java
import java.util.Arrays;
class Solution {

    
    public long solution(int n, int[] times) {
        long result = 0;
        Arrays.sort(times);
        long left =0, right = (long)n*times[times.length-1];
		
		while(left <= right) {
		
			long mid = (left+right)/2;
			long sum = 0;
			for(int i=0; i <times.length; i++) {
				sum += mid/times[i];
			}
			
			
			if(sum>=n) {
				result = mid;
				right= mid-1;
			}else{
				left = mid+1;
			}
			
		}
		
		
        return result;
    }
}